File: 2022/main.cpp
#include <algorithm>
#include <fstream>
#include <iostream>
#include <numeric>
#include <sstream>
#include <string>
#include <vector>

using namespace std;

class Data
{
};

int part1( Data data )
{
    int result = 0;
    return result;
}

int part2( Data data )
{
    int result = 0;
    return result;
}

Data parse( istream& is )
{
    auto data = Data();
    return data;
}

int main( int argc, char* argv[] )
{
    for(int i = 1; i < argc; ++i)
    {
        ifstream fh( *++argv  );
        auto data = parse( fh );
        cout << "part 1: " << part1( data ) << "\n";
        cout << "part 2: " << part2( data ) << "\n";
    }

    return 0;
}

File: 2022/main.py
# /usr/bin/env python

import argparse
import collections
import copy
import functools
import itertools
import re
from typing import *


class Data:
    pass


def parse(lines: List[str]) -> Data:
    return Data()


def part1(data: Data) -> int:
    result = 0
    return result


def part2(data: Data) -> int:
    result = 0
    return result


if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument("filename")
    args = parser.parse_args()

    with open(args.filename) as fh:
        data = parse(fh.readlines())

    p1 = part1(copy.deepcopy(data))
    print(f"Part 1: {p1}")
    p2 = part2(copy.deepcopy(data))
    print(f"Part 2: {p2}")
